### PR TITLE
improvement(GCP): Add a general label for GCE instances

### PR DESCRIPTION
**Pull request by roydahan**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster_gce.py [hunk 1]
```diff
@@ -347,6 +347,8 @@ def set_tags_as_labels(_instance):
 
         self.log.info('Created %s instance %s', 'spot' if spot else 'on-demand', instance)
         try:
+            general_tag = {"created_by": "sct"}
```
Comment (fgelcer) [inline sdcm/cluster_gce.py:350]: the `test_id` will also be very useful

**Comment by fruch** [comment]
@roydahan what are you trying to solve here ?

sct resource should be tagged with something similar already

I've added it in:
https://github.com/scylladb/scylla-cluster-tests/pull/4981

(and someone was asking why it's needed 😜 )

File: sdcm/cluster_gce.py (hunk 1 above)
Comment (roydahan) [reply sdcm/cluster_gce.py:350]: We already have it.
This is to tag something as default for the cases the bottom one fails.

**Comment by roydahan** [comment]
> @roydahan what are you trying to solve here ?
> 
> sct resource should be tagged with something similar already
> 
> I've added it in: #4981
> 
> (and someone was asking why it's needed 😜 )

As I explained you, something that will always tag instances, regardless of the checks and values we have.

**Comment by fruch** [comment]
@roydahan 

I think this would do a better job, the previous code didn't handle all the problematic values, and not the length which is 63 max
this replace anything that isn't (`a-z` or `_-`) to `_`

```diff
diff --git a/sdcm/cluster_gce.py b/sdcm/cluster_gce.py
index 9536f428..037a8412 100644
--- a/sdcm/cluster_gce.py
+++ b/sdcm/cluster_gce.py
@@ -12,6 +12,7 @@
 # Copyright (c) 2020 ScyllaDB

 import os
+import re
 import time
 import logging
 from typing import Dict, Any, ParamSpec, TypeVar
@@ -297,9 +298,14 @@ class GCECluster(cluster.BaseCluster):  # pylint: disable=too-many-instance-attr
     def _create_instance(self, node_index, dc_idx, spot=False, enable_auto_bootstrap=False):
         def set_tags_as_labels(_instance):
             self.log.debug(f"Expected tags are {self.tags}")
-            lower_tags = dict((k.lower(), v.lower().replace('.', '-')) for k, v in self.tags.items())
-            self.log.debug(f"Lower tags are {lower_tags}")
-            self._gce_services[dc_idx].ex_set_node_labels(_instance, lower_tags)
+            # https://cloud.google.com/resource-manager/docs/tags/tags-creating-and-managing#adding_tag_values
+            regex_key = re.compile(r'[^a-z-_]')
+            def to_short_name(value):
+                return regex_key.sub('_', value.lower())[:60]
+
+            normalized_tags = {to_short_name(k): to_short_name(v) for k, v in self.tags.items()}
+            self.log.debug(f"normalized tags are {normalized_tags}")
+            self._gce_services[dc_idx].ex_set_node_labels(_instance, normalized_tags)
```

**Comment by roydahan** [comment]
@fruch you could send a PR and that's it :)

**Comment by roydahan** [comment]
Labels are bit ugly now, but at least work:

Screen Shot 2022-09-15 at 19 53 05
Screen Shot 2022-09-15 at 19 52 57

**Comment by fruch** [comment]
> Labels are bit ugly now, but at least work:
> 
> Screen Shot 2022-09-15 at 19 53 05
> Screen Shot 2022-09-15 at 19 52 57
> 
> 

This seems problematic, especially the test id, I've forgot the numbers here, I think..

**Comment by fgelcer** [comment]
> This seems problematic, especially the test id, I've forgot the numbers here, I think..
very problematic without the numbers

File: sdcm/cluster_gce.py [hunk 2]
```diff
@@ -297,9 +298,15 @@ def _get_disks_struct(self, name, dc_idx):
     def _create_instance(self, node_index, dc_idx, spot=False, enable_auto_bootstrap=False):
         def set_tags_as_labels(_instance):
             self.log.debug(f"Expected tags are {self.tags}")
-            lower_tags = dict((k.lower(), v.lower().replace('.', '-')) for k, v in self.tags.items())
-            self.log.debug(f"Lower tags are {lower_tags}")
-            self._gce_services[dc_idx].ex_set_node_labels(_instance, lower_tags)
+            # https://cloud.google.com/resource-manager/docs/tags/tags-creating-and-managing#adding_tag_values
+            regex_key = re.compile(r'[^a-z-_]')
```
Comment (fruch) [inline sdcm/cluster_gce.py:302]: adding the number would make it a bit more reasonable:
```suggestion
            regex_key = re.compile(r'[^a-z0-9-_]')
```

**Comment by roydahan** [comment]
Screen Shot 2022-09-21 at 18 46 10

**Pull request merged**